examples/sui-move/simple_warrior/sources/example.move

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Demonstrates wrapping objects using the `Option` type.
+module simple_warrior::example {
+    use std::option::{Self, Option};
+    use sui::object::{Self, UID};
+    use sui::tx_context::TxContext;
+
+    struct Sword has key, store {
+        id: UID,
+        strength: u8,
+    }
+
+    struct Warrior has key {
+        id: UID,
+        sword: Option<Sword>,
+    }
+
+    public fun new_sword(strength: u8, ctx: &mut TxContext): Sword {
+        Sword { id: object::new(ctx), strength }
+    }
+
+    public fun new_warrior(ctx: &mut TxContext): Warrior {
+        Warrior { id: object::new(ctx), sword: option::none() }
+    }
+
+    public fun equip(warrior: &mut Warrior, sword: Sword): Option<Sword> {
```

Comment:
To build the PTB, you first check whether the warrior has the sword or not, if it does, then you build a PTB that calls `option::destroy_some` on the result and then transfers the resulting `Sword`.  If not, you simply call `option::destroy_none` on the result. (So the conditional on whether the sword exists or not is part of the TypeScript or Rust that creates the PTB rather than the PTB itself).

Reply:
Sure, we can structure it that way!  This will require some larger changes in the accompanying docs, but it probably is easier to understand, and it still illustrates the point about wrapping in and out of `Option`.